### DISCUSSION TITLE
Extending Pocketbase (JS) and filtering a collection that has JSON

### DISCUSSION THREAD

**Discussion by TheHypnoo** [pocketbase/pocketbase#5622] [Q&A] [answered]
Hello, sorry for the inconvenience.

I'm trying to extend Pocketbase with the options it offers, in this case with Javascript. I have an existential doubt, in this case, I have a collection called: Diagrams, in that case, inside I have a field called zones, and that is an array of objects: [ {..}, {...} ]. I would like to know if it is possible and how to do the findRecordsByFilter taking into account that a field is a json.

Forgive me if I have made myself clear. If you have any questions, let me know. 

Thanks in advance.

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase supports dot-notation lookups for `json` field types (assuming that `zones` is a `json` field) which under the hood utilizes the builtin SQLite [`json_extract` function](https://www.sqlite.org/json1.html#jex).

This works fine with objects, but for array of objects you need to know the index of the targetd object in order the filter to work because there is no wildcard support, aka. you can have something like:
```js
// valid (targets the first object)
zones.0.someField = "test"

// invalid
zones.*.someField = "test"
```

Support for wildcard json array of objects filter exists somewhere burried in my todo list but remains a very low priority.

A better approach would be to store your json data in a separate collection that can reference the main one with a `relation` field (this will also allow simple indexes on the individual json object columns).

Also, this is not PocketBase specific, but note that storing large amount of data in a single record is known to cause performance issues (e.g. https://github.com/pocketbase/pocketbase/discussions/3859#discussioncomment-7744045) and I'd recommend to split your data properly across dedicated collections.

**Reply by TheHypnoo** [reply]
All right, to make the work easier, what I did was to create a separate collection and a relationship in between.

For this case, what is the easy and correct way to get from the collection ‘zones’ the filtering of sensors (which is a relation) in this case is an array. I have the id of a sensor and I want to filter to only get all the zones where this id is. 

I have not found a suitable way for this use case. Sorry for the inconvenience and thanks.

**Comment by Tonku** [comment]
" storing large amount of data" any clue on what size of a json would make it considered as large json ? 1MB ?

**Reply by ganigeorgiev** [reply]
It is difficult to say and it will vary from one use case to another.
Personally I would consider anything more than 5KB a "large" JSON field value.
